Gems/Atom/RPI/Code/Source/RPI.Reflect/Material/MaterialTypeAsset.cpp

```diff
@@ -82,23 +82,13 @@ namespace AZ
             {
                 return true;
             }
-            for (auto& shaderItem : m_generalShaderCollection)
+            if (!m_generalShaderCollection.InitializeShaderOptionGroups())
             {
-                if (!shaderItem.InitializeShaderOptionGroup())
-                {
-                    return false;
-                }
+                return false;
             }
-
             for (auto& materialPipelinePair : m_materialPipelinePayloads)
             {
-                for (auto& shaderItem : materialPipelinePair.second.m_shaderCollection)
-                {
-                    if (!shaderItem.InitializeShaderOptionGroup())
-                    {
-                        return false;
-                    }
-                }
+                materialPipelinePair.second.m_shaderCollection.InitializeShaderOptionGroups();
             }
```

Comment:
This part lost its return if the function call failed

Reply:
LEt me check on that.